pkg/migrations/sql/idempotent/base.sql

```diff
@@ -2350,6 +2358,239 @@ SET search_path = pg_temp;
 REVOKE ALL ON FUNCTION SCHEMA_CATALOG.delete_series_from_metric(text, bigint[])FROM PUBLIC;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.delete_series_from_metric(text, bigint[]) to prom_modifier;
 
+DO $block$
+BEGIN
+    IF SCHEMA_CATALOG.is_timescaledb_installed() AND SCHEMA_CATALOG.get_timescale_major_version() >= 2 THEN
```

Comment:
nit: since all these blocks are the same if, do we want to wrap all these functions in one block? instead of a block-per-function

Reply:
I could go either way on this. I did it this way so each function was in a standalone "chunk" of code, but I don't feel strongly either way. I'll switch.